WordPress/src/main/java/org/wordpress/android/ui/media/MediaBrowserActivity.java

```diff
@@ -74,18 +75,20 @@
     private BroadcastReceiver mReceiver = new BroadcastReceiver() {
         @Override
         public void onReceive(Context context, Intent intent) {
-            String action = intent.getAction();
-            if (ConnectivityManager.CONNECTIVITY_ACTION.equals(action)) {
+            if (ConnectivityManager.CONNECTIVITY_ACTION.equals(intent.getAction())) {
                 // Coming from zero connection. Continue what's pending for delete
-                String blogId = String.valueOf(WordPress.getCurrentBlog().getLocalTableBlogId());
-                if (WordPress.wpDB.getMediaDeleteQueueItems(blogId).getCount() > 0) {
-                    startMediaDeleteService();
+                Blog currentBlog = WordPress.getCurrentBlog();
+
+                if (currentBlog != null) {
```

Comment:
This NPE fix looks good. Note, though, that you could skip it entirely by using `WordPress.getCurrentLocalTableBlogId()`. And if you don't mind one more suggestion, I recommend using a more optimal way to detect whether there are pending deletes (since this receiver may fire quite frequently). Ex:

```
private BroadcastReceiver mReceiver = new BroadcastReceiver() {
    @Override
    public void onReceive(Context context, Intent intent) {
        if (ConnectivityManager.CONNECTIVITY_ACTION.equals(intent.getAction())) {
            int blogId = WordPress.getCurrentLocalTableBlogId();
            if (blogId != -1 && WordPress.wpDB.hasMediaDeleteQueueItems(blogId)) {
                startMediaDeleteService();
            }
        }
    }
};
```

Where `WordPress.wpDB.hasMediaDeleteQueueItems` is:

```
public boolean hasMediaDeleteQueueItems(int blogId) {
    return SqlUtils.boolForQuery(db,
            "SELECT 1 FROM " + MEDIA_TABLE + " WHERE uploadState=? AND blogId=?",
            new String[]{"delete", Integer.toString(blogId)});
}
```